csrc/torch_memory_saver.cpp

```diff
@@ -118,32 +119,32 @@ struct _AllocationMetadata {
     size_t size;
     CUdevice device;
     CUmemGenericAllocationHandle allocHandle;
+    std::string tag;
 };
 
 class TorchMemorySaver {
 public:
     TorchMemorySaver() {}
 
-    cudaError_t malloc(void **ptr, size_t size) {
+    cudaError_t malloc(void **ptr, size_t size, const std::string& tag = "default") {
```

Comment:
nit: I personally like to make it non-default arg, b/c we always pass in something from upper level, but anyway this is just a nit